reviewrot/basereview.py

```diff
@@ -1,8 +1,41 @@
 from collections import OrderedDict
+import datetime
+from dateutil.relativedelta import relativedelta
 
 
 class BaseService(object):
 
+    def format_reviews(self, username, title, url, comments, created_at,
+                       state_, value, duration):
+        """
+        find the relative time difference between now and
+        review request filed to retrieve relative information
+        """
+        rel_diff = relativedelta(datetime.datetime.now(),
+                                 created_at)
+        """
+        find the absolute time difference between now and
+        review request filed to retrieve absolute information
+        """
+        abs_diff = datetime.datetime.now() - created_at
+        if (state_ is not None and value is not None and
```

Comment:
I'm envisioning this a bit differently. Rather than the `BaseService` class contain logic about how to format/interpret a review, I think the `BaseReview`/`GithubReview`/etc. classes should. Here's how I think the general structure could be (not tested in any way):

Ex: From within `GithubService`
```python
reviews = []
for pr in pull_requests:
    review = GithubReview(repo, user=pr.user.login, title=pr.title,
                          url=pr.html_url, comments=pr.comments,
                          created_at=pr.created_at)
    if review.matches_criteria(state=state_, duration=duration, value=value):
        reviews.append(review)
```

Within `BaseReview` you can calculate `abs_diff` and `rel_diff` against `datetime.now()` upon instantiation.